OPS-18782 Add Connector API documentation for missing AvailabilityBlock fields (#641)
MewsSystems/gitbook-connector-api: operations/availabilityblocks.md

            "FirstTimeUnitStartUtc": "2021-10-14T00:00:00Z",
            "LastTimeUnitStartUtc": "2021-10-17T00:00:00Z",
            "ReleasedUtc": "2021-10-13T00:00:00Z",

>             "RollingReleaseOffset": "P-3DT4H",

Reviewer: If the two release fields are optional and "mutually exclusive", doesn't that mean that only one of them will be present in the JSON? The example now shows both, which violates that rule.

| `UpdatedUtc` | string | required | Last update date and time of the block in UTC timezone in ISO 8601 format. |
| `FirstTimeUnitStartUtc` | string | required | Start of the time interval, expressed as the timestamp for the start of the first time unit, in UTC timezone ISO 8601 format. See [Time units](../concepts/time-units.md). |
| `LastTimeUnitStartUtc` | string | required | End of the time interval, expressed as the timestamp for the start of the last time unit, in UTC timezone ISO 8601 format. See [Time units](../concepts/time-units.md). |

> | `ReleasedUtc` | string | optional | The moment when the block and its availability is released in UTC timezone in ISO 8601 format. Mutually exclusive with `RollingReleaseOffset`; the block will not be automaticaly released if neither `ReleasedUtc` nor `RollingReleaseOffsetUtc` is specified. |

Reviewer: I think the change is perfectly well described here. However, I am concerned that changing `ReleasedUtc` from `required` to `optional` is a potentially breaking change. Has this been agreed with the API Team?

Reviewer 2: Answered here: https://github.com/MewsSystems/gitbook-connector-api/pull/641/files/5cc5cd3ac52eb8b54ae60bb82ff7d9f06e970caf#r1745350934

> | `RollingReleaseOffset` | optional | Exact offset from the start of availability adjustments to the moment the availability adjustment should be released, in ISO 8601 duration format. Mutually exclusive with `ReleasedUtc`; the block will not be automaticaly released if neither `ReleasedUtc` nor `RollingReleaseOffsetUtc` is specified. |

Reviewer 3: ```suggestion
| `RollingReleaseOffset` | string | optional | Exact offset from the start of availability adjustments to the moment the availability adjustment should be released, in ISO 8601 duration format. Mutually exclusive with `ReleasedUtc`; the block will not be automaticaly released if neither `ReleasedUtc` nor `RollingReleaseOffsetUtc` is specified. |
```

> | `RollingReleaseOffset` | string | optional | Exact offset from the start of availability adjustments to the moment the availability adjustment should be released, in ISO 8601 duration format. Mutually exclusive with `ReleasedUtc`; the block will not be automaticaly released if neither `ReleasedUtc` nor `RollingReleaseOffsetUtc` is specified. |

Reviewer 4: "not specified": does this mean that they are not sent, or does it mean they contain the value "null"?
It also states RollingReleaseOffset AND RollingReleaseOffsetUtc.
Going from not required to optional is a breaking change.

Reviewer: I would interpret the two cases exactly the same, setting a property to "null" is the same as not including it, as "null" is not a value (@jnv correct me if I'm wrong). I've also raised the other two points myself, but it's good to hear it directly from an API user, thank you @remywrevcontrol. @jvn please consider these points, especially about the breaking change.

Reviewer 2: `optional` means "can be null".

We are aware this is a breaking change and our partner success manager is already in touch with you regarding this unfortunate change.

From documentation perspective, this is a breaking change. From OpenAPI Spec perspective, the property was never marked as required in the first place which is why this wasn't flagged as a breaking change on our watch. But it really started to be null only once the domain team introduced rolling releases earlier this year.

This is a f*up on our part and we'll have a long discussion with the team responsible, unfortunately we can't do much else now other than to fix the documentation and help you to accommodate for these changes.

| `Name` | string | optional | The name of the block. |
| `FirstTimeUnitStartUtc` | string | required | Start of the time interval, expressed as the timestamp for the start of the first time unit, in UTC timezone ISO 8601 format. See [Time units](../concepts/time-units.md). |
| `LastTimeUnitStartUtc` | string | required | End of the time interval, expressed as the timestamp for the start of the last time unit, in UTC timezone ISO 8601 format. See [Time units](../concepts/time-units.md). |

> | `ReleasedUtc` | string | optional | The moment when the block and its availability is released, in UTC timezone ISO 8601 format. |

Reviewer: Do we need to add the same statement about being mutually exclusive to the fields here like you did with `Get all availability`, or does it not apply in this case?

Reviewer 2: Added: "Takes precedence over `RollingReleaseOffset`."

| `BookerId` | [Guid update value](_objects.md#string-update-value) | optional | Unique identifier of the Booker as a creator of an availability block (or `null` if not updated). |
| `Notes` | [String update value](_objects.md#string-update-value) | optional | Additional notes of the block (or `null` if not updated). |
| `Budget` | [Currency](accountingitems.md#currency-value) update value | optional | The tentative budget for the total price of reservations (or `null` if not updated). |
| `RollingReleaseOffset` | [String update value](_objects.md#string-update-value) | optional | Exact offset from the start of availability adjustments to the moment the availability adjustment should be released, in ISO 8601 duration format. Required if `ReleaseStrategy` is set to `RollingRelease`, ignored otherwise. |
| `ReleasedUtc` | [String update value](_objects.md#string-update-value) | optional | The moment when the block and its availability is released, in UTC timezone ISO 8601 format. Required if `ReleaseStrategy` is set to `FixedRelease`, or used when `ReleaseStrategy` update is unspecified. |

> | `ReleaseStrategy` | [String update value](_objects.md#string-update-value) for [Release strategy](#release-strategy) | optional | The strategy for automatic release of the availability block (or `null` if not updated). |

Reviewer: ```suggestion
| `ReleaseStrategy` | [String update value](_objects.md#string-update-value) | optional | The strategy for automatic release of the availability block. If this property is not specified, the release strategy is not updated. |
```

#### Release strategy

The strategy for automatic release of the availability block.

- `FixedRelease` - The availability block is released at a fixed time.

> - `RollingRelease` - Each availability ajdustment is released at a fixed offset from its start.

Reviewer: ```suggestion
- `RollingRelease` - Each availability adjustment is released at a fixed offset from its start.
```
